src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/transformers/DeduplicatingResourceTransformer.kt

```diff
@@ -0,0 +1,116 @@
+package com.github.jengelman.gradle.plugins.shadow.transformers
+
+import com.github.jengelman.gradle.plugins.shadow.internal.WithPatternFilterable
+import java.io.File
+import java.nio.ByteBuffer
+import java.security.MessageDigest
+import javax.inject.Inject
+import org.apache.tools.zip.ZipOutputStream
+import org.gradle.api.GradleException
+import org.gradle.api.file.FileTreeElement
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * Transformer to include files with identical content only once in the shadow JAR.
+ *
+ * Multiple files with the same path but different content lead to an error.
+ */
+@CacheableTransformer
+public open class DeduplicatingResourceTransformer
```

Comment:
Part of a duplicate of #1548.

Reply:
I realized that `PropertiesFileTransformer` may not be suitable for reproducible builds, as it relies on `j,u.Hashtable` (the order of map entries isn't deterministic). Let me see what can be done about that.